### ISSUE TITLE
Check the tabIndex of commonly tabbed to elements to ensure that they remain accessible

### ISSUE THREAD

**Issue by danspratling** [jackdomleo7/Checka11y.css#8]
## Proposal
Check the `tabIndex` of commonly tabbed to elements to ensure that they remain accessible.

## Problem
Sometimes, elements which should be able to be tabbed to have their tabIndex incorrectly set, meaning they are skipped over when navigating using the keyboard

```
<button type="submit" tabIndex="-1">Submit</button>
```
The above button would be skipped past when navigating using the keyboard, making its associated form un-submittable.

## Solution
For a specified list of elements, if a tabIndex is present and equal to "-1" an error should be shown.

Elements:
- [ ] a[href]
- [ ] area[href]
- [ ] input:not([disabled])
- [ ] select:not([disabled])
- [ ] textarea:not([disabled])
- [ ] button:not([disabled])
- [ ] iframe
- [ ] [contentEditable=true]

Any others which I've missed?

**Comment by jackdomleo7** [comment]
I like this feature! If I'm correct, using `tabIndex="-1"` means you can only focus on that element programmatically?

**Comment by danspratling** [comment]
So just to confirm

- [ ] Error for the above elements if `tabIndex="-1"`
- [ ] Warn on any other element if tabindex is set at all?

**Comment by jackdomleo7** [comment]
Maybe the second one could be a different feature? But yes, I think that would be good.

**Comment by Manaswini1832** [comment]
Hey Jack! I'd like to work on the negative tabIndex issue. Could you please assign me to it?
